src/ParametricCliqStateMachine.jl

```diff
@@ -0,0 +1,337 @@
+  # add messages as priors to this sub factor graph
+  msgfcts = DFGFactor[]
+  svars = DFG.getVariableIds(subfg)
+  for (msym, belief) = (msgs.belief)
+    if msym in svars
+      # TODO prior missing manifold information? is it not available in variable?
+      # TODO add MsgPrior parametric type
+      @warn "TODO I just packed in MvNormal and ignored cov"
+      msgPrior =  MsgPrior(MvNormal(belief.val[:,1], 1), belief.inferdim)
+      fc = addFactor!(subfg, [msym], msgPrior, autoinit=false)
+      push!(msgfcts, fc)
+    end
+  end
+  return msgfcts
+end
+
+"""
+    $SIGNATURES
+
+Notes
+- Parametric state machine function nr. 2
+"""
+function waitForUp_ParametricStateMachine(csmc::CliqStateMachineContainer)
+
+  infocsm(csmc, "Par-2, wait for up messages of needed")
+
+  setCliqDrawColor(csmc.cliq, "purple") #TODO don't know if this is correct color
+  csmc.drawtree ? drawTree(csmc.tree, show=false, filepath=joinpath(getSolverParams(csmc.dfg).logpath,"bt.pdf")) : nothing
+
+  for e in Graphs.out_edges(csmc.cliq, csmc.tree.bt)
+    @info "$(csmc.cliq.index): take! on edge $(e.index)"
+    # Blocks until data is available.
+    beliefMsg = take!(csmc.tree.messages[e.index].upMsg)
+    @info "$(csmc.cliq.index): Belief message recieved with status $(beliefMsg.status)"
+    #TODO save up message (and add priors to cliqSubFg) gaan eers in volgende stap kyk hoe dit werk
+    #kies csmc vir boodskappe vir debugging, dis 'n vector een per kind knoop
+    push!(csmc.parametricMsgsUp, beliefMsg)
+  end
+
+  return solveUp_ParametricStateMachine
+end
+
+function Graphs.in_edges(vert::V, gr::GenericIncidenceList{V, Edge{V}, Vector{V}}) where {V}
+  inclist = gr.inclist
+  targid = vert.index
+  inlist = Edge{V}[]
+  for edgelist in inclist
+    for ed in edgelist
+      if ed.target.index == targid
+        push!(inlist, ed)
+      end
+    end
+  end
+  return inlist
+end
+
+"""
+    $SIGNATURES
+
+Notes
+- Parametric state machine function nr. 3
+"""
+function solveUp_ParametricStateMachine(csmc::CliqStateMachineContainer)
+
+  infocsm(csmc, "Par-3, Solving Up")
+
+  setCliqDrawColor(csmc.cliq, "red")
+  csmc.drawtree ? drawTree(csmc.tree, show=false, filepath=joinpath(getSolverParams(csmc.dfg).logpath,"bt.pdf")) : nothing
+
+  # is dit nodig om vir parametric die factors by te sit?
+  #TODO maybe change to symbols
+  msgfcts = DFGFactor[]
+  for upmsgs in csmc.parametricMsgsUp
+    append!(msgfcts, addMsgFactors!(csmc.cliqSubFg, upmsgs))
+  end
+
+  # store the cliqSubFg for later debugging
+  opts = getSolverParams(csmc.dfg)
+  if opts.dbg
+    DFG.saveDFG(csmc.cliqSubFg, joinpath(opts.logpath,"logs/cliq$(csmc.cliq.index)/fg_beforeupsolve"))
+    drawGraph(csmc.cliqSubFg, show=false, filepath=joinpath(opts.logpath,"logs/cliq$(csmc.cliq.index)/fg_beforeupsolve.pdf"))
+  end
+
+  #TODO UpSolve cliqSubFg here
+  # slaap om somme te simileer
+  sleep(rand()*2)
```

Comment:
this is where we call the JuliaNLSolvers routines to do the actual matrix computations on cliqSubFg -- or better yet just a call to batchSolveParametric!(csmc.cliqSubFg) as you had suggested before.

Reply:
I think it will work to start with. I like the idea of `batchSolveParametric!(csmc.cliqSubFg)` since you can also use it to solve bigger factor graphs  and then do the incremental in the tree.

Reply:
We can also look at computing the conditionals

Reply:
Computing the conditionals will definitely be of interest, actually expect @tonioteran might be interested in that.  There is another name in the lingo called "smart factors" (L. Carlone et al.)